src/nativeMain/kotlin/util/Util.kt

```diff
@@ -31,4 +33,21 @@ fun Display.sendClientMessage(type: Atom, window: Window, propagate: Boolean = f
 }
 
 fun Display.sendClientMessage(type: Atom, propagate: Boolean = false, eventMask: Long = NoEventMask, vararg data: Long) =
-    sendClientMessage(type, rootWindow, propagate, eventMask, *data)
\ No newline at end of file
+    sendClientMessage(type, rootWindow, propagate, eventMask, *data)
+
+fun Window.setCursor(dpy: Display, cur: UInt = 2.toUInt(), window: Window = this) { 
+    var cursorInitialized = windowCursorInitialized[window]
+    if (cursorInitialized != true) {
+        windowCursors.put(window, XCreateFontCursor(dpy.ptr, cur))
+        XDefineCursor(dpy.ptr, window, windowCursors[window]!!)
+        windowCursorInitialized.put(window, true)
+    }
+}
+fun Window.unsetCursor(dpy: Display, window: Window = this) {
```

Comment:
You could also remove `window` argument and replace usages of it with `this`. Since if you have access to an instance of a window you can use this extension method directly